### PR TITLE
Sanitize library names before running liblist command

### PR DESCRIPTION
**Pull request by chrjorgensen**
### Changes

This PR will fix the error described in issue #1672.
It will also fix errors in other library list commands:
- show an error when changing current library, if the new current library name starts with `#`
- don't allow non-existing libraries with name starting with `#` when changing the library list

### Checklist

* [x] have tested my change
* [x] eslint is not complaining

### CODE REVIEW

File: src/views/LibraryListView.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { ConnectionConfiguration, GlobalConfiguration } from "../api/Configurati
 import { instance } from "../instantiate";
 import { t } from "../locale";
 import { Library as LibraryListEntry } from "../typings";
+import { Tools } from "../api/Tools";
```
Comment (worksofliam) [inline src/views/LibraryListView.ts:6]: Unused import was added.

**Review by worksofliam — CHANGES_REQUESTED** [review]

**Review by worksofliam — CHANGES_REQUESTED** [review]
I'd consider this a high impact change since it is used by a lot of our extension (and others!).

The tests are passing and things are working as normal.

Other than the unused import, the code and functionality works good.

Optionally, if you felt inclined, I'd love to see a new test case to show that it works with libraries that start with `#`.

File: src/views/LibraryListView.ts (hunk 1 above)
Comment (chrjorgensen) [reply src/views/LibraryListView.ts:6]: Ups - a leftover from when I thought I should sanitize library names in this module. Fixed!

**Comment by chrjorgensen** [comment]
@worksofliam I'm currently working on some tests - which revealed some defects with the handling of '$'!
We have IT auditing (💩) at work tomorrow, but I hope to have some fixes ready tomorrow evening or soon after...

**Comment by chrjorgensen** [comment]
@worksofliam I fixed errors for sanitizing names with '$' and validating library list with sanitized names - and finally added some tests to verify library name sanitizing. Please have another go...

**Review by worksofliam — APPROVED** [review]
Looking good @chrjorgensen! Thanks for your work here.

**Pull request merged**